src/Kovan-DssSpell.t.sol

```diff
@@ -1278,71 +1672,116 @@ contract DssSpellTest is DSTest, DSMath {
         assertTrue(totalGas <= 8 * MILLION);
     }
 
-    function test_nextCastTime() public {
-        hevm.warp(1606161600); // Nov 23, 20 UTC (could be cast Nov 26)
-
+    function testSpellIsCast_new_End() public {
         vote(address(spell));
-        spell.schedule();
+        scheduleWaitAndCast(address(spell));
+        assertTrue(spell.done());
 
-        uint256 monday_1400_UTC = 1606744800; // Nov 30, 2020
-        uint256 monday_2100_UTC = 1606770000; // Nov 30, 2020
+        // Contracts set
+        assertEq(end.vat(), address(vat));
+        assertEq(end.cat(), address(cat));
+        assertEq(EndAbstractNew(address(end)).dog(), address(dog));
+        assertEq(end.vow(), address(vow));
+        assertEq(end.pot(), address(pot));
+        assertEq(end.spot(), address(spotter));
 
-        // Day tests
-        hevm.warp(monday_1400_UTC);                                    // Monday,   14:00 UTC
-        assertEq(spell.nextCastTime(), monday_1400_UTC);               // Monday,   14:00 UTC
+        assertEq(esm.end(), address(end));
 
-        if (spell.officeHours()) {
-            hevm.warp(monday_1400_UTC - 1 days);                       // Sunday,   14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
 
-            hevm.warp(monday_1400_UTC - 2 days);                       // Saturday, 14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+        // Authorization
+        bytes32[] memory ilks = IlkRegistryAbstract(addr.addr("ILK_REGISTRY")).list();
+        for (uint256 i = 0; i < ilks.length; i++) {
+            FlipAbstract flip = FlipAbstract(IlkRegistryAbstract(addr.addr("ILK_REGISTRY")).flip(ilks[i]));
 
-            hevm.warp(monday_1400_UTC - 3 days);                       // Friday,   14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC - 3 days);  // Able to cast
+            assertEq(flip.wards(address(end)), 1);
+            assertEq(flip.wards(address(end_old)), 0);
+        }
 
-            hevm.warp(monday_2100_UTC);                                // Monday,   21:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC + 1 days);  // Tuesday,  14:00 UTC
+        assertEq(vat.wards(address(end)), 1);
+        assertEq(vat.wards(address(end_old)), 0);
+        assertEq(cat.wards(address(end)), 1);
+        assertEq(cat.wards(address(end_old)), 0);
+        assertEq(dog.wards(address(end)), 1);
+        assertEq(vow.wards(address(end)), 1);
+        assertEq(vow.wards(address(end_old)), 0);
+        assertEq(pot.wards(address(end)), 1);
+        assertEq(pot.wards(address(end_old)), 0);
+        assertEq(spotter.wards(address(end)), 1);
+        assertEq(spotter.wards(address(end_old)), 0);
+
+        assertEq(end.wards(address(esm)), 1);
+        assertEq(end_old.wards(address(esm_old)), 1);
+    }
```

Comment:
I would also include a full ESM attack shutdown integration test for extra safety.

Reply:
Yes, already added to checklist as tests to be implemented.